### PR TITLE
Stash troublesome modification

### PR DESCRIPTION
**Pull request by shnarazk**

### CODE REVIEW

File: src/assign/learning_rate.rs [hunk 1]
```diff
@@ -25,55 +25,48 @@ impl ActivityIF<VarId> for AssignStack {
...
 impl Var {
+    // Note: `update_rewards` should be called befero `restart`
+    // Therefore timestamp must be smaller than t anytime.
+    // But conflict_analyze can assert a new var. So we can't expect v.timestamp < self.num_conflict
     fn update_activity(&mut self, t: usize, decay: f64, reward: f64) -> f64 {
         if self.timestamp < t {
             let duration = (t - self.timestamp) as f64;
+            let rate = self.participated as f64 / duration as f64;
             self.reward *= decay.powf(duration);
-            self.reward += reward;
```
Comment (shnarazk) [inline src/assign/learning_rate.rs:76]: This is the bug.

**Comment by shnarazk** [comment]
# Comparios of ar reward decays

|  # |     var decay rate =   |    0.94  |    0.95  |   0.06  |   0.97  |    0.98 |     0.99 |
----:|------------------------|---------:|---------:|--------:|--------:|--------:|---------:|
|  1 | "UF250(100)           |  104.833 |   91.768 |  83.714 |   99.530 |   99.623 |   96.800 |
|  2 | "UUF250(100)          |  234.147 |  229.626 | 223.844 |  233.310 |  244.023 |  265.446 |
|  3 | "3/SAT/v360-c1530/002 |   19.396 |  119.504 | 143.509 |    1.293 |  137.315 |   71.038 |
|  4 | "3/SAT/v360-c1530/030 |  432.097 |  356.021 | 164.724 |  427.761 |  242.154 |  109.252 |
|  5 | "3/SAT/v360-c1530/033 |   81.504 |  128.113 |  32.766 |   23.343 |   52.083 |    0.560 |
|  6 | "3/SAT/v360-c1530/035 |    8.846 |  115.351 |  36.333 |   64.843 |    0.781 |    3.093 |
|  7 | "3/SAT/v360-c1530/039 |  877.174 |  962.971 | 209.038 |  109.431 |  501.466 |  836.190 |
|  8 | "3/SAT/v360-c1530/051 |  396.875 |  414.104 | 440.537 |  110.758 |  512.382 |  560.171 |
|  9 | "3/SAT/v360-c1530/060 |    4.003 |  805.867 |  21.592 |  711.342 |  109.136 |  953.822 |
| 10 | "3/SAT/v360-c1530/073 |   79.622 |   12.721 |  89.035 |  341.513 |  118.155 |   21.292 |
| 11 | "3/SAT/v360-c1530/087 |    6.040 |    3.700 |  14.555 |    3.676 |    7.487 |   90.158 |
| 12 | "3/SAT/v360-c1530/093 |    8.725 |  139.435 | 328.051 |  135.761 |   74.257 |   63.303 |
| 13 | "3/UNS/v360-c1530/001 |  240.774 |  354.140 | 338.357 |  361.578 |  293.571 |  361.043 |
| 14 | "3/UNS/v360-c1530/015 |  180.216 |  171.043 | 178.518 |  213.459 |  172.971 |  200.454 |
| 15 | "3/UNS/v360-c1530/028 |  310.589 |  347.362 | 267.943 |  312.344 |  281.932 |  339.812 |
| 16 | "3/UNS/v360-c1530/029 |  508.590 |  444.609 | 418.977 |  491.018 |  347.357 |  389.885 |
| 17 | "3/UNS/v360-c1530/031 |  263.836 |  239.324 | 243.031 |  300.348 |  329.567 |  293.931 |
| 18 | "3/UNS/v360-c1530/053 |  285.493 |  253.496 | 289.577 |  246.384 |  293.242 |  309.549 |
| 19 | "3/UNS/v360-c1530/061 |  539.641 |  513.805 | 622.417 |  607.377 |  561.572 |  729.725 |
| 20 | "3/UNS/v360-c1530/086 |  224.600 |  221.826 | 156.411 |  171.550 |  203.394 |  187.913 |
| 21 | "3/UNS/v360-c1530/089 |  282.482 |  225.852 | 240.985 |  422.556 |  330.392 |  231.562 |
| 22 | "3/UNS/v360-c1530/096 |  166.093 |  176.679 | 158.742 |  176.858 |  191.041 |  143.867 |
| 23 | "SR2015/itox          |    1.758 |    1.883 |   2.104 |    2.033 |    1.780 |    1.877 |
| 24 | "SR2015/m283          |   23.197 |   50.615 |  46.632 |   43.260 |   45.037 |   31.685 |
| 25 | "SR2015/38b           |   22.483 |   23.224 |   6.814 |   16.259 |    7.580 |    3.478 |
| 26 | "SR2015/44b           |  254.409 |  429.112 | 483.027 |  243.316 |  276.895 |  210.483 |
|    |                       | 5557.423 | 6832.151 | 5241.23 | 5870.901 | 5435.193 | 6506.389 |

**Comment by shnarazk** [comment]
| no staging | 2760.67 |
| backbone ||
| highhalf ||

**Comment by shnarazk** [comment]
# Revise clause reduction scheme

### 3SAT360

- basic: 8730.430
- 0.5.3
- new 4594.653
